### DISCUSSION TITLE
pyvista & trame - Add Actor to rendered plotter

### DISCUSSION THREAD

**Discussion by mcaox** [pyvista/pyvista#5203] [Q&A] [answered]
Good Afternoon,

Both Trame & Pyvista are very exciting concepts ... 
We are exploring these technologies for a possible migration of an application.

We try to develop some enhanced "simple app" to implement features this application has.

These "enhanced  simple applications" are based on the tutorial [Base script](https://tutorial.pyvista.org/tutorial/09_trame/a_trame_simple.html#sphx-glr-tutorial-09-trame-a-trame-simple-py)

The environnement is the following :

- trame==3.3.0
- trame-client==2.12.6
- trame-server==2.12.1
- trame-vtk==2.6.2
- trame-vuetify==2.3.1
- pyvista==0.42.3

We have the following questions :

- **Agenda for the migration to vue and impacts on pyvista (within Trame) use**
    - The as if example does not work.
        - "JS Error => error: TypeError: r.a.extend is not a function" 
    - We adapted it this way.
        - Should also vue3 become the default for pyvista in comming December ?

          ```
          from trame.widgets import vuetify3 as vuetify
          from trame.ui.vuetify3 import SinglePageLayout
          
          [...]
          
          server = get_server(client_type="vue3")
          ``` 

- **Adding Actor to the current plotter**

   - Both button and callback are added

          ```
          [...]
          def add_actor():
              pl.add_mesh(pv.Sphere())
          
          with SinglePageLayout(server) as layout:
              with layout.content:
                  vuetify.VBtn("add",click=add_actor)
                  view = plotter_ui(pl)
          [...]
          ```

   - The front loose the connection and the server raises this exception
      - What does the exception really mean ?
      - What would be the proper approach to add an actor to the rendered plotter ?
      ```
      JS Error => Malformed message:  {"wslink": "1.0", "id": "p
      ...
       ["addViewProp", ["instance:${000001afc6e6e190}"]], ["addLight", ["instance:${000001afc6682300}"]], ["addLight", ["instance:${000001afc6681c20}"]], ["addLight", ["instance:${000001afc6682b40}"]], ["addLight", ["instance:${000001afc66821a0}"]], ["addLight", ["instance:${000001afc6681960}"]]]}], "calls": [["addRenderer", ["instance:${000001afc615cd30}"]]], "mtime": 2645, "extra": {"vtkRefId": "000001afc62248f0", "centerOfRotation": [0.0, 9.751734167337418, 3.491544723510742], "camera": "000001afc61bfce0"}}}}
      ```

**Comment by banesullivan** [comment]
cc @jourdain

**Reply by mcaox** [reply, marked as answer]
As we had the same error with vtk we opened an issue on [trame](https://github.com/Kitware/trame/discussions/376)

**Reply by jourdain** [reply]
It might be due to the release of aiohttp 3.9. That was released last week. Try to fix the version to 3.8.6 and see if the problem disappears.

**Reply by jourdain** [reply]
An independent tester reported that forcing aiohttp to 3.8.6 fixed the issue.
